lib/syncmap/syncmap.go

```diff
@@ -0,0 +1,40 @@
+package syncmap
+
+import "sync"
+
+type SyncMap[K comparable, V any] struct {
+	_map *sync.Map
+}
+
+func New[K comparable, V any]() SyncMap[K, V] {
+	return SyncMap[K, V]{
+		_map: &sync.Map{},
+	}
+}
+
+func (sm SyncMap[K, V]) Set(key K, value V) {
+	sm._map.Store(key, value)
+}
+
+func (sm SyncMap[K, V]) GetWithStatus(key K) (value V, ok bool) {
```

Comment:
how about just return nil if Get returns nothing

Reply:
you have to return the type's zero value (which isn't always nil), which is what we do here when there is no value in the map. The reason why maps have `value, bool` as return values is so you can distinguish between the key not existing in the map vs the key existing but holding the zero value. 

`Get(key K) (value V)`  is right below for scenarios where you only want the value

Reply:
I'd rather you make this `Has` then. Or rename this Get and have the current Get be `MustGet`.

Reply:
we can rename the two if you prefer the 2 value return to be `Get` instead of the single value return. I don't know that one really has priority over the other but I made single return have the shorter name since is returns fewer things. making a separate Has func doesn't follow the map form and makes it less of a drop in replacement

Reply:
`gin.Context` also has the `Get` return 2 values, with `MustGet` so it seems fine

https://github.com/gin-gonic/gin/blob/master/context.go#L261-L269

Reply:
basically map has the form where you can just use the value if it exists or the zero value `f(m[key])` and the double assignment form `v, ok := m[key]`, which is more verbose. So we want drop in replacements for each of these.

Reply:
what do you think of `Get(key K) (value V, ok bool)` and `GetValue(key K) (value V)` instead of `MustGet` which sounds like it might panic if it doesn't exist?

Reply:
go calls these `mapaccess1(t *maptype, h *hmap, key unsafe.Pointer) unsafe.Pointer` and `mapaccess2(t *maptype, h *hmap, key unsafe.Pointer) (unsafe.Pointer, bool)` internally

https://github.com/golang/go/blob/master/src/runtime/map.go#L396-L457

Reply:
ok here go calls these `Lookup(key string) (value string, ok bool)` and `Get(key string) string`, updated to use these names

Reply:
> although in our case we don't want to panic if it doesn't exist, we just want the zero value. So the Must could be misleading

I slightly think that if you call Get, you do have the assumption it must exist. But nbd